src/DocumentFormat.OpenXml.Linq/Feature/RootXElementFeature.cs

```diff
@@ -25,67 +25,115 @@ public RootXElementFeature(OpenXmlPart part)
             _events.Change += OnPartRootChange;
         }
 
-        /// <summary>
-        /// Gets or sets the root LINQ to XML element.
-        /// </summary>
-        [DisallowNull]
-        public XElement? Root
+        /// <inheritdoc />
+        public XDocument Document
         {
             get
             {
-                if (_rootXElement is null)
+                if (_partXDocument is not null)
                 {
-                    if (_part.IsRootElementLoaded)
+                    if (_partXDocument.Root is null)
                     {
-                        _rootXElement = XElement.Parse(_part.RootElement.OuterXml);
-                    }
-                    else
-                    {
-                        _rootXElement = LoadRootXElementFromStream();
+                        // Add null or the root XElement.
+                        _partXDocument.Add(LoadRootXElement());
                     }
+
+                    return _partXDocument;
                 }
 
-                return _rootXElement;
+                _partXDocument = LoadRootXElement() switch
+                {
+                    // The part is empty.
+                    null => new XDocument(new XDeclaration("1.0", "UTF-8", "yes")),
+
+                    // The part has XML content that was loaded from the part stream.
+                    { Document: { } } root => root.Document!,
+
+                    // The part has XML content that was loaded from RootElement.OuterXml.
+                    var root => new XDocument(new XDeclaration("1.0", "UTF-8", "yes"), root),
+                };
+
+                return _partXDocument;
             }
 
             set
             {
-                _rootXElement = value ?? throw new ArgumentNullException(nameof(value));
+                if (value.Root is null)
+                {
+                    throw new ArgumentException("Root must not be null.");
+                }
+
+                _partXDocument = value;
 
                 Save();
             }
         }
 
+        /// <inheritdoc />
+        [DisallowNull]
+        public XElement? Root
```

Comment:
Do we need this on the feature itself? Or would an extension method be sufficient?

Reply:
For a public interface, I would want to have it in the interface. If we made the interface internal, an extension method would be sufficient. Here is why:

I currently see two congruent sets of properties and methods in the public API:

1. `Root`, `GetXElement`, `SetXElement`, and `SaveXElement`; and
2. `Document`, `GetXDocument`, `SetXDocument`, `SaveXDocument`.

Removing just the `Root` property would break that congruence. If we made the interface internal, the public API would only have the following methods:

1. `GetXElement`, `SetXElement`, and `SaveXElement`; and
2. `GetXDocument`, `SetXDocument`, `SaveXDocument`.

That would also be totally fine and possibly even better than the first set. In any case, we should make sure the public API provides two congruent sets of properties (optional) and methods (required).

Should we want to make the `IPartRootXElementFeature` interface (current naming) internal, we would also make the `OpenXmlLinqExtensions.GetPartRootXElementFeature` extension method internal.

Reply:
I'm fine either way here. Just thinking of what is really being abstracted behind the interface and what would ideally be a default implementation if we could use that with the SDK.

Reply:
See my question above. I suggest we make the interface internal.